client/clientutil/modelinfo.go

```diff
@@ -0,0 +1,431 @@
+				// will be snap names which are required to be short
+				fmt.Fprintf(w, "  - %s\n", headerStringElem)
+			}
+
+		// timestamp needs to be formatted in an identical manner to how fmtTime works
+		// from timeMixin package in cmd/snap
+		case "timestamp":
+			timestamp, ok := headerValue.(string)
+			if !ok {
+				return formatInvalidTypeErr(headerName)
+			}
+
+			// parse the time string as RFC3339, which is what the format is
+			// always in for assertions
+			t, err := time.Parse(time.RFC3339, timestamp)
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(w, "timestamp:\t%s\n", fmtTime(t, opts.AbsTime))
+
+		// long string key we don't want to rewrap but can safely handle
+		// on "reasonable" width terminals
+		case "device-key-sha3-384":
+			// also flush the writer before continuing so the previous keys
+			// don't try to align with this key
+			w.Flush()
+			headerString, ok := headerValue.(string)
+			if !ok {
+				return formatInvalidTypeErr(headerName)
+			}
+
+			switch {
+			case opts.TermWidth > 86:
+				fmt.Fprintf(w, "device-key-sha3-384: %s\n", headerString)
+			case opts.TermWidth > 66:
+				fmt.Fprintln(w, "device-key-sha3-384: |")
+				strutil.WordWrapPadded(w, []rune(headerString), "  ", opts.TermWidth)
+			}
+		case "snaps":
+			// also flush the writer before continuing so the previous keys
+			// don't try to align with this key
+			w.Flush()
+			snapsHeader, ok := headerValue.([]interface{})
+			if !ok {
+				return formatInvalidTypeErr(headerName)
+			}
+			if len(snapsHeader) == 0 {
+				// unexpected why this is an empty list, but just ignore for
+				// now
+				continue
+			}
+
+			fmt.Fprintf(w, "snaps:\n")
+			if err := printVerboseSnapsList(w, snapsHeader); err != nil {
+				return err
+			}
+
+		// long base64 key we can rewrap safely
+		case "device-key":
+			headerString, ok := headerValue.(string)
+			if !ok {
+				return formatInvalidTypeErr(headerName)
+			}
+			// the string value here has newlines inserted as part of the
+			// raw assertion, but base64 doesn't care about whitespace, so
+			// it's safe to replace the newlines
+			headerString = strings.ReplaceAll(headerString, "\n", "")
+			fmt.Fprintln(w, "device-key: |")
+			strutil.WordWrapPadded(w, []rune(headerString), "  ", opts.TermWidth)
+
+		// The rest of the values should be single strings
+		default:
+			headerString, ok := headerValue.(string)
+			if !ok {
+				return formatInvalidTypeErr(headerName)
+			}
+			fmt.Fprintf(w, "%s:\t%s\n", headerName, headerString)
+		}
+	}
+	return w.Flush()
+}
+
+// PrintModelAssertion will format the provided serial or model assertion based on the parameters given in
+// YAML format, or serialize it raw if Assertion is set. The output will be written to the provided io.Writer.
+func PrintModelAssertion(w *tabwriter.Writer, modelFormatter ModelFormatter, modelAssertion asserts.Model, serialAssertion *asserts.Serial, opts PrintModelAssertionOptions) error {
```

Comment:
modelFormatter should go just before opts in the arg list

Reply:
Fixed!